### PR TITLE
service: Update function tables

### PR DESCRIPTION
**Pull request by VolcaEM**
This was based on Switchbrew pages:

- NIM: https://switchbrew.org/wiki/NIM_services
- NPNS: https://switchbrew.org/wiki/NPNS_services
- NS: https://switchbrew.org/wiki/NS_Services
- PL_U: https://switchbrew.org/wiki/Shared_Database_services
- NVDRV: https://switchbrew.org/wiki/NV_services
- NVMEMP: https://switchbrew.org/wiki/NV_services
- PCIe: https://switchbrew.org/wiki/PCIe_services
- PCV: https://switchbrew.org/wiki/PCV_services
- PM: https://switchbrew.org/wiki/Process_Manager_services
- PRepo: https://switchbrew.org/wiki/BCAT_services
- PSC: https://switchbrew.org/wiki/PSC_services
- PSM: https://switchbrew.org/wiki/PTM_services
- SM (controller): https://switchbrew.org/wiki/IPC_Marshalling
- NSD: https://switchbrew.org/wiki/Sockets_services
- SFDNSRes: https://switchbrew.org/wiki/Sockets_services
- SPL: https://switchbrew.org/wiki/SPL_services
- Time: https://switchbrew.org/wiki/PSC_services
- USB: https://switchbrew.org/wiki/USB_services
- VI: https://switchbrew.org/wiki/Display_services
- WLAN: https://switchbrew.org/wiki/WLAN_services

I'm not too sure about SPL since there is a number in the FunctionInfo which is two different services, maybe this requires "splitting" SPL in multiple FunctionInfo(s)

We should also find better names for NIM's RequestSyncTicketLegacy (old version) and RequestSyncTicket (new version)

### CODE REVIEW

**Comment by ogniK5377** [comment]
Clang format please

File: src/core/hle/service/nim/nim.cpp [hunk 1]
```diff
@@ -121,11 +121,83 @@ class NIM final : public ServiceFramework<NIM> {
             {39, nullptr, "PrepareShutdown"},
             {40, nullptr, "ListApplyDeltaTask"},
             {41, nullptr, "ClearNotEnoughSpaceStateOfApplyDeltaTask"},
-            {42, nullptr, "Unknown1"},
-            {43, nullptr, "Unknown2"},
-            {44, nullptr, "Unknown3"},
-            {45, nullptr, "Unknown4"},
-            {46, nullptr, "Unknown5"},
+            {42, nullptr, "Unknown42"},
+            {43, nullptr, "Unknown42"},
```
Comment (superheri) [inline src/core/hle/service/nim/nim.cpp:125]: Copy paste? Shouldn't it be Unknown43?

Comment (VolcaEM) [reply src/core/hle/service/nim/nim.cpp:125]: Yes, it was a typo, thanks

File: src/core/hle/service/sm/controller.cpp [hunk 2]
```diff
@@ -47,13 +47,14 @@ void Controller::QueryPointerBufferSize(Kernel::HLERequestContext& ctx) {
...
 
+// https://switchbrew.org/wiki/IPC_Marshalling
 Controller::Controller() : ServiceFramework("IpcController") {
     static const FunctionInfo functions[] = {
-        {0x00000000, &Controller::ConvertSessionToDomain, "ConvertSessionToDomain"},
-        {0x00000001, nullptr, "ConvertDomainToSession"},
-        {0x00000002, &Controller::DuplicateSession, "DuplicateSession"},
+        {0x00000000, &Controller::ConvertCurrentObjectToDomain, "ConvertCurrentObjectToDomain"},
+        {0x00000001, nullptr, "CopyFromCurrentDomain"},
+        {0x00000002, &Controller::CloneCurrentObject, "CloneCurrentObject"},
```
Comment (ogniK5377) [inline src/core/hle/service/sm/controller.cpp:55]:
```suggestion
        {0, &Controller::ConvertCurrentObjectToDomain, "ConvertCurrentObjectToDomain"},
        {1, nullptr, "CopyFromCurrentDomain"},
        {2, &Controller::CloneCurrentObject, "CloneCurrentObject"},
        {3, &Controller::QueryPointerBufferSize, "QueryPointerBufferSize"},
        {4, &Controller::CloneCurrentObjectEx, "CloneCurrentObjectEx"},
```

File: src/core/hle/service/nfp/nfp.cpp [hunk 3]
```diff
@@ -55,7 +55,7 @@ class IUser final : public ServiceFramework<IUser> {
             {19, &IUser::GetState, "GetState"},
             {20, &IUser::GetDeviceState, "GetDeviceState"},
             {21, &IUser::GetNpadId, "GetNpadId"},
-            {22, &IUser::GetApplicationAreaSize, "GetApplicationAreaSize"},
+            {22, &IUser::GetApplicationArea2, "GetApplicationArea2"},
```
Comment (ogniK5377) [inline src/core/hle/service/nfp/nfp.cpp:58]: Why was this changed? It's not GetApplicationArea2

File: src/core/hle/service/nim/nim.cpp [hunk 4]
```diff
@@ -181,7 +256,8 @@ class NIM_SHP final : public ServiceFramework<NIM_SHP> {
             {305, nullptr, "RequestCreateVirtualAccount"},
             {306, nullptr, "RequestDeviceLinkStatus"},
             {400, nullptr, "GetAccountByVirtualAccount"},
-            {500, nullptr, "RequestSyncTicket"},
+            {401, nullptr, "GetVirtualAccount"},
+            {500, nullptr, "RequestSyncTicketLegacy"},
```
Comment (ogniK5377) [inline src/core/hle/service/nim/nim.cpp:260]: Legacy?

File: src/core/hle/service/ns/ns.cpp [hunk 5]
```diff
@@ -567,14 +567,14 @@ class NS_DEV final : public ServiceFramework<NS_DEV> {
...
-            {4, nullptr, "GetShellEventHandle"},
+            {4, nullptr, "GetShellEvent"},
             {5, nullptr, "GetShellEventInfo"},
             {6, nullptr, "TerminateApplication"},
             {7, nullptr, "PrepareLaunchProgramFromHost"},
-            {8, nullptr, "LaunchApplication"},
-            {9, nullptr, "LaunchApplicationWithStorageId"},
-            {10, nullptr, "TerminateApplication2"},
-            {11, nullptr, "GetRunningApplicationProcessId"},
+            {8, nullptr, "LaunchApplicationFromHost"},
```
Comment (ogniK5377) [inline src/core/hle/service/ns/ns.cpp:574]: Why was this also renamed?

File: src/core/hle/service/ns/ns.cpp [hunk 6]
```diff
@@ -567,14 +567,14 @@ class NS_DEV final : public ServiceFramework<NS_DEV> {
             {0, nullptr, "LaunchProgram"},
             {1, nullptr, "TerminateProcess"},
             {2, nullptr, "TerminateProgram"},
-            {4, nullptr, "GetShellEventHandle"},
+            {4, nullptr, "GetShellEvent"},
```
Comment (ogniK5377) [inline src/core/hle/service/ns/ns.cpp:570]: Why was this also changed?

File: src/core/hle/service/nim/nim.cpp (hunk 4 above)
Comment (VolcaEM) [reply src/core/hle/service/nim/nim.cpp:260]: 500 is called "RequestSyncTicketLegacy" because it isn't used "anymore" and SwitchBrew sometimes uses "Old", sometimes uses "2", and sometimes uses "Legacy" to rename "older versions" of functions

"RequestSyncTicketLegacy" (500) is "RequestSyncTicket" but for versions 1.0.0 through 8.1.0, and there is already a "RequestSyncTicket" (503) which is the "new" version introduced in version 9.0.0
